jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -721,8 +721,10 @@ private InterpreterResult executeSql(String propertyKey, String sql,
         statement = connection.createStatement();
 
         // fetch n+1 rows in order to indicate there's more rows available (for large selects)
-        statement.setFetchSize(getMaxResult());
-        statement.setMaxRows(maxRows);
+        statement.setFetchSize(Integer.parseInt(interpreterContext
+                .getLocalProperties().getOrDefault("limit", "" + getMaxResult())));
```

Comment:
nit: I think this pattern of int -> string -> int is a bit odd, perhaps helper func can help here
`Integer.parseInt(interpreterContext
                .getLocalProperties().getOrDefault("limit", "" + getMaxResult()`

Reply:
fixed